### DISCUSSION TITLE
Caching Vercel-built Assets

### DISCUSSION THREAD

**Discussion by johnrichardrinehart** [vercel/community#625] [Help] [answered]
![image](https://user-images.githubusercontent.com/6321578/170668131-91612e43-4837-437f-a72b-3e862079eb3f.png)
Above is a Vercel-support-provided screenshot of some of our most-downloaded assets over the past ~7 days. The bandwidth that we're using (and being charged) for Vercel-generated assets like the WASM is kind of insane. I'd like to architect our application so that we're serving these files from a cache that we manage, so that we can control costs a little bit better. Vercel has recommended that [they **don't recommend**](https://vercel.com/support/articles/using-cloudflare-with-vercel) using Cloudflare to proxy and cache traffic.

So, how can I accomplish this in a way that doesn't violate Vercel's recommended best practices?

**Comment by Cl3arglass** [comment, marked as answer]
Looks like somehow your project is importing some modules which includes a wasm binary and it is getting bundled. The wasm binary in question includes fn signatures which seem related with https://github.com/emurgo/cardano-serialization-lib. Checking on your most recent deployment I do see the library mentioned: https://vercel.com/jpgstore/wagmi/FScDQNfSuPLj3aU8SKT4A6HfeH9q
You may absolutely introduce a Cloudflare proxy, the biggest warning we have is that with a proxy in between Vercel we cannot guarantee the best possible performance we can offer for your site. https://vercel.com/support/articles/can-i-use-a-proxy-on-top-of-my-vercel-deployment#global-distribution

**Reply by johnrichardrinehart** [reply]
Thanks, Holden! Great to see you here!
